tap_hubspot/__init__.py

```diff
@@ -836,6 +859,13 @@ def get_selected_streams(remaining_streams, ctx):
             selected_streams.append(stream)
     return selected_streams
 
+def has_selected_custom_field(mdata):
+    top_level_custom_props = [x for x in mdata if len(x) == 2 and 'property_' in x[1]]
+    for prop in top_level_custom_props:
+        if mdata.get(prop, {}).get('selected') == True:
+            return True
+    return False
+
```

Comment:
Move this up to above `sync_deals`, just for order-s sake (even though Python doesn't care)